### PR TITLE
Add reading AWS root/config endpoint

### PR DESCRIPTION
**Pull request by joelthompson**
Previously, there was no read listener on the AWS config/root endpoint,
so clients couldn't read any of the config options that were set. This
adds a read operation so that all configured values can be read, except
for the secret_key, which is filtered out and never returned.

Fixes https://groups.google.com/forum/#!msg/vault-tool/0CBEdDbSEoI/9NDKkxM_FAAJ

### CODE REVIEW

File: builtin/logical/aws/path_config_root.go [hunk 1]
```diff
@@ -50,6 +51,35 @@ func pathConfigRoot(b *backend) *framework.Path {
...
+
+	if entry != nil {
+		if err := entry.DecodeJSON(&config); err != nil {
+			return nil, err
+		}
+	}
+
+	configData := map[string]interface{}{
+		"access_key":   config.AccessKey,
+		"secret_key":   "",
```
Comment (kalafut) [inline builtin/logical/aws/path_config_root.go]: Simply omitting the key entirely is the more common pattern.

File: builtin/logical/aws/path_config_root.go [hunk 2]
```diff
@@ -50,6 +51,35 @@ func pathConfigRoot(b *backend) *framework.Path {
...
+	if err != nil {
+		return nil, err
+	}
+	var config rootConfig
+
+	if entry != nil {
+		if err := entry.DecodeJSON(&config); err != nil {
+			return nil, err
+		}
+	}
```
Comment (kalafut) [inline builtin/logical/aws/path_config_root.go]: If `entry == nil` I think we should return `nil, nil`.

Comment (joelthompson) [reply builtin/logical/aws/path_config_root.go]: done

File: builtin/logical/aws/path_config_root.go (hunk 1 above)
Comment (joelthompson) [reply builtin/logical/aws/path_config_root.go]: I was mostly worried about people who expect to get the same keys back on reading data as they input in writing to the endpoints, but if this is the more common pattern, makes sense to just omit it. Done.

**Comment by kalafut** [comment]
Thanks. Now we can probably improve:  https://github.com/lawliet89/terraform-provider-vault/blob/8cf62e16d302fc5b4fe5d4c04896730976ba2349/vault/resource_aws_secret_backend.go#L166

**Pull request merged**